spec/dummy/spec/requests/posts_requests_spec.rb

```diff
@@ -55,19 +55,37 @@
     end
 
     context 'with valid params' do
-      it 'should return current user posts' do
-        user = create(:user)
-        user1 = create(:user_1)
-
-        user_posts = create_list(:post, 2, user_id: user.id)
-        create_list(:post, 2, user_id: user1.id)
-
-        access_token = jwt_and_refresh_token(user, 'user')[0]
-
-        get '/posts', headers: { 'Authorization': "Bearer #{access_token}" }
+      context 'as user' do
+        let(:user) { create(:user) }
+        let(:user1) { create(:user_1) }
+        let(:access_token) { jwt_and_refresh_token(user, 'user')[0] }
+
+        it 'should return current user posts' do  
+          user_posts = create_list(:post, 2, user_id: user.id)
+          create_list(:post, 2, user_id: user1.id)
+  
+          get '/posts', headers: { 'Authorization': "Bearer #{access_token}" }
+  
+          expect(response).to have_http_status(200)
+          expect(parsed_response.map { |p| p['id'] }).to match_array(user_posts.map(&:id))
+        end
+      end
 
-        expect(response).to have_http_status(200)
-        expect(parsed_response.map { |p| p['id'] }).to match_array(user_posts.map(&:id))
+      context 'as admin' do
```

Comment:
Please add a spec with invalid admin token in `with invalid params` context above. Any one spec (either with blacklisted / old access token) is enough to make sure it is validating the second resource in the list.